src/Android/MainActivity.cs

```diff
@@ -278,6 +297,40 @@ private void ListenYubiKey(bool listen)
             }
         }
 
+        private async Task<bool> ListenU2F(Dictionary<string, object> data)
+        {
+            var appId = Uri.Parse((string)data["appId"]);
+            var challenge = Encoding.ASCII.GetBytes((string) data["challenge"]);
+            var keys = ((JArray) data["keys"]).ToObject<List<Dictionary<string, string>>>();
+            var registeredKeys = new List<RegisteredKey>();
+            foreach (var key in keys)
+            {
+                var keyHandle = new KeyHandle(
+                    Encoding.ASCII.GetBytes(key["keyHandle"]),
```

Comment:
Might be the same, a "url safe" base 64 encoded string

Reply:
any updates?

Reply:
No, I haven't found any documentation on how the Original Android API is supposed to work with other providers than Google itself. There's also no Open Source projects using either the Xamarin Library or the original Android Java classes to access third-party U2F/FIDO2 providers. I'm pretty much stuck not knowing what I might do wrong/what format the Android API expects.

Reply:
I am not really in this topic, but why doesn't https://github.com/android/security-samples/tree/master/Fido helps / is an option?

Reply:
Took a look at this? https://fidoalliance.org/specs/fido-uaf-v1.2-rd-20171128/fido-uaf-client-api-transport-v1.2-rd-20171128.html#org.fidoalliance.intent.fido_operation-intent